animated-transformer/src/lib/gtensor/gtensor.ts

```diff
@@ -941,14 +949,43 @@ export function makeRange<T extends DName>(
   start: number,
   end: number,
   step: number,
-  dtype: 'float32' | 'int32' = 'float32'
+  dtype: 'float32' | 'int32' = 'float32',
 ): GTensor<T> {
   return new GTensor<T>(tf.range(start, end, step, dtype), [dname]);
 }
 
+/* Returns a 2D triangular mask of shape [size, size]
```

Comment:
Comment nit: this isn't necessarily a mask, it's just a square matrix with all values where x > y having a given constant, and all others having a different constant.